packages/auth/src/AuthProvider.tsx

```diff
@@ -113,52 +140,67 @@ type AuthProviderState = {
  *  </AuthProvider>
  * ```
  */
-export class AuthProvider extends React.Component<
-  AuthProviderProps,
-  AuthProviderState
-> {
-  static defaultProps = {
-    skipFetchCurrentUser: false,
-  }
-
-  state: AuthProviderState = {
-    loading: true,
-    isAuthenticated: false,
-    userMetadata: null,
-    currentUser: null,
-    hasError: false,
-  }
-
-  rwClient: AuthClient
-
-  constructor(props: AuthProviderProps) {
-    super(props)
-    this.rwClient = createAuthClient(
+export const AuthProvider: React.FC<PropsWithChildren<AuthProviderProps>> = (
+  props: PropsWithChildren<AuthProviderProps>
+) => {
```

Comment:
Will ditch `React.FC`. Personally, I advocate for `PropsWithChildren` since it allows you to inherit the full gamut of renderable children from the react types package without needing to sort through what your given version of React supports. (This is of course assuming the types package is working that given week, as React 18 has shown us isn't quite a given 😓)

That said, can revert both if you'd rather.

Reply:
Just found this comment from Dan Abramov https://stackoverflow.com/a/71809927/88106
> I want to note that you absolutely don't _have_ to use this `PropsWithChildren` helper. (It is primarily useful for the [codemod](https://github.com/eps1lon/types-react-codemod#implicit-children), not manual usage.)

So I say we get rid of it here in our code 🙂

Reply:
@Tobbe Done

Reply:
Thanks for fixing it so quickly. And I'm really sorry it's taking me so long to do a full review and approval of this PR. I'll get to it next week. Picky swear.